src/components/views/messages/MKeyVerificationRequest.tsx

```diff
@@ -15,187 +15,71 @@ limitations under the License.
 */
 
 import React from "react";
-import { MatrixEvent, User } from "matrix-js-sdk/src/matrix";
-import { logger } from "matrix-js-sdk/src/logger";
-import {
-    canAcceptVerificationRequest,
-    VerificationPhase,
-    VerificationRequestEvent,
-} from "matrix-js-sdk/src/crypto-api";
+import { MatrixEvent } from "matrix-js-sdk/src/matrix";
 
-import { MatrixClientPeg } from "../../../MatrixClientPeg";
 import { _t } from "../../../languageHandler";
 import { getNameForEventRoom, userLabelForEventRoom } from "../../../utils/KeyVerificationStateObserver";
-import { RightPanelPhases } from "../../../stores/right-panel/RightPanelStorePhases";
 import EventTileBubble from "./EventTileBubble";
-import AccessibleButton from "../elements/AccessibleButton";
-import RightPanelStore from "../../../stores/right-panel/RightPanelStore";
+import { useMatrixClientContext } from "../../../contexts/MatrixClientContext";
 
-interface IProps {
+interface Props {
     mxEvent: MatrixEvent;
     timestamp?: JSX.Element;
 }
 
-export default class MKeyVerificationRequest extends React.Component<IProps> {
-    public componentDidMount(): void {
-        const request = this.props.mxEvent.verificationRequest;
-        if (request) {
-            request.on(VerificationRequestEvent.Change, this.onRequestChanged);
-        }
-    }
-
-    public componentWillUnmount(): void {
-        const request = this.props.mxEvent.verificationRequest;
-        if (request) {
-            request.off(VerificationRequestEvent.Change, this.onRequestChanged);
-        }
-    }
-
-    private openRequest = (): void => {
-        let member: User | undefined;
-        const { verificationRequest } = this.props.mxEvent;
-        if (verificationRequest) {
-            member = MatrixClientPeg.safeGet().getUser(verificationRequest.otherUserId) ?? undefined;
-        }
-        RightPanelStore.instance.setCards([
-            { phase: RightPanelPhases.RoomSummary },
-            { phase: RightPanelPhases.RoomMemberInfo, state: { member } },
-            { phase: RightPanelPhases.EncryptionPanel, state: { verificationRequest, member } },
-        ]);
-    };
+interface MKeyVerificationRequestContent {
+    body?: string;
+    format?: string;
+    formatted_body?: string;
+    from_device: string;
+    methods: Array<string>;
+    msgtype: "m.key.verification.request";
+    to: string;
+}
 
-    private onRequestChanged = (): void => {
-        this.forceUpdate();
-    };
+const MKeyVerificationRequest: React.FC<Props> = ({ mxEvent, timestamp }) => {
```

Comment:
I'm not against a bit of documentation about the component. What it is it's purpose and it should behave :)

Reply:
Thanks, good call.